doc/tutorial/read_mapper/read_mapper_step2.cpp

```diff
@@ -0,0 +1,92 @@
+#include <seqan3/argument_parser/all.hpp>
+#include <seqan3/io/sequence_file/input.hpp>
+#include <seqan3/io/stream/debug_stream.hpp>
+#include <seqan3/range/view/get.hpp>
+
+using namespace seqan3;
+//! [solution]
+//! [data_storage_t]
+struct data_storage_t
+{
+    concatenated_sequences<std::string> ref_id;
+    concatenated_sequences<std::vector<dna5>> ref_seq;
+    concatenated_sequences<std::string> query_id;
+    concatenated_sequences<std::vector<dna5>> query_seq;
+    concatenated_sequences<std::vector<phred42>> query_qual;
+};
+//! [data_storage_t]
+
+//! [read_input]
+void read_input(std::filesystem::path const & reference_path,
+                std::filesystem::path const & query_path,
+                data_storage_t & storage)
+//! [read_input]
+{
+    sequence_file_input reference_in{reference_path};
+    storage.ref_id  = std::move(get<field::ID>(reference_in));
+    storage.ref_seq = std::move(get<field::SEQ>(reference_in));
+    debug_stream << "Reference IDs: " << storage.ref_id << '\n';
+
+    sequence_file_input query_in{query_path};
+    storage.query_id   = std::move(get<field::ID>(query_in));
+    storage.query_seq  = std::move(get<field::SEQ>(query_in));
+    storage.query_qual = std::move(get<field::QUAL>(query_in));
+    debug_stream << "Number of queries: " << std::ranges::size(storage.query_id) << '\n';
+}
+
+void run_program(std::filesystem::path const & reference_path,
+                 std::filesystem::path const & query_path,
+                 std::filesystem::path const & sam_path,
+                 uint8_t const errors)
+{
+    data_storage_t storage{};
+    read_input(reference_path, query_path, storage);
+
+    (void) sam_path;
+    (void) errors;
```

Comment:
remove those

Reply:
I need them.

We need to also add `-Werror=unused-parameter`